### ISSUE TITLE
Query not happening with query key function

### ISSUE THREAD

**Issue by tibotiber** [tannerlinsley/react-query#267]
Hello,

I'm upgrading from 0.3.23 to 1.0.29 and I'm facing with something that looks like a bug.

I have a custom hook as follow:
```js
export const useGqlAnonymousQuery = (qvFn, options) => {
  return useQuery(
    () => {
      const [query, variables] = qvFn()
      return [getQueryKey(query), variables]
    },
    async (key, variables) => {
      return getClient().request(print(qvFn()[0]), variables)
    },
    options
  )
}
```

Basically it uses a query key function to modify the string part of the key. This used to work but now the query function doesn't get called at all and suspense goes in error boundary saying data is undefined.

I've tried replacing the query key function with an example value that get returned (e.g. `['test', {a:2}]`) and it works. Then I've tried with a super simple query key function (`() => ['test', {a:2}]`) and it fails as described previously.

My workaround is to compute the modified key before calling useQuery, which is fine, but I prefer to report this in case it's a bug. 

Note: I'm using suspense mode (not sure if it happens without suspense). 

Another quick one, it seems the new signature of the query function is `queryFn: Function(key, variables) => Promise(data/error)` but the README is still at `queryFn: Function(variables) => Promise(data/error) `. Is that right?

Except for that, it's a pretty neat upgrade, the new options allow for a more declarative usage and have cut by half my custom hook code. The devtools is a superb addition as well. Great job Tanner! 👍

**Comment by tannerlinsley** [comment]
Could you perhaps throw this into a codesandbox for us?

**Comment by tibotiber** [comment]
Sure, sorry for not doing that earlier. 

I've tried to reproduce in a sandbox and am not able to. The closest I've got is https://codesandbox.io/s/quizzical-montalcini-f7ns8. It's based on your suspense sandbox. I've changed the versions of react, react-dom and react-query to match mine to be sure, I've removed the experimental react stuff since I downgraded to 16.8.6. And of course I've added a (rather contrived) custom hook that's a minimal version of mine. However everything still works as expected in the sandbox.

I've spent hours on this but I'm not able to isolate a simple repro case. My code is large and private, and you probably don't want to dive in anyway 😉.

Overall, on my end it works as if suspense mode is ignored and that the suspending doesn't happen. I'm sure suspense is set to true. The query does not appear in the react-query devtools before the error happens. I've checked the source to understand what may happen and it makes no sense. Really not sure that to do more.

If it can't be reproduced and is not reported by anyone else and I have a workaround, we might as well close this (?). Let me know, and thanks for your time in any case.
